aten/src/ATen/native/cuda/UnfoldBackwardKernel.cu

```diff
@@ -0,0 +1,204 @@
+#include <ATen/native/UnfoldBackward.h>
+
+#include <ATen/native/cuda/Loops.cuh>
+#include <ATen/cuda/detail/OffsetCalculator.cuh>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/native/cuda/zmath.cuh>
+
+#include <vector>
+
+// Note on naming: it is unconventional.
+// grad_in does not mean that it is a gradient wrt to input,
+// grad_in/grad_out is just an input/output of unfold_backward kernel.
+//
+// unfold_backward, the algorithm is described in
+// /native/cpu/UnfoldBackwardKernel.cpp
+
+namespace at { namespace native {
+
+namespace {
+
+template <int n_threads, int n_elems_per_thread, typename func_t>
+C10_LAUNCH_BOUNDS_2(n_threads, n_elems_per_thread)
+__global__ void _unfold_backward_elementwise_kernel(int total_n_elems, func_t f) {
+  constexpr int total_work_block = n_threads * n_elems_per_thread;
+  int idx = total_work_block * blockIdx.x + threadIdx.x;
+
+  #pragma unroll
+  for (int i = 0; i < n_elems_per_thread; ++i) {
+    if (idx < total_n_elems) {
+      f(idx);
+      idx += n_threads;
+    }
+  }
+}
+
+template <int n_threads, int n_elems_per_thread, typename func_t>
+static void _launch_unfold_backward_kernel(int total_n_elems, func_t f) {
+  TORCH_INTERNAL_ASSERT(
+    total_n_elems >= 0 && total_n_elems <= std::numeric_limits<int32_t>::max()
```

Comment:
nit: `_unfold_backward_internal_kernel` already handle the case where total_n_elems==0. So you could change this to check `>0` and remove the if statement below I think.

Reply:
paying tribute to `legacy::launch_kernel` here ;)